arch/arm/core/fatal.c

```diff
@@ -73,6 +73,23 @@ void z_arm_fatal_error(unsigned int reason, const z_arch_esf_t *esf)
 		esf_dump(esf);
 	}
 #endif /* CONFIG_EXCEPTION_DEBUG */
+
+	/* LOG the IRQn that was unhandled */
+#if defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE) && defined(CONFIG_LOG)
```

Comment:
This looks like it should work as-is on armv6m as well.


https://developer.arm.com/documentation/ddi0419/c/System-Level-Architecture/System-Level-Programmers--Model/Registers/The-special-purpose-program-status-registers--xPSR

I'm sure there is something similar available in armvN-A as well.

Reply:
I don't have access to armv6m to test that, would you like me to speculatively add support for v6?

Is this code covered by Zephyr's CI at runtime?

Reply:
@SebastianBoe `__get_IPSR()` is used in [`isr_wrapper.c:72`](https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/arm/core/cortex_m/isr_wrapper.c#L72).

Reply:
ok, seems safe then :)